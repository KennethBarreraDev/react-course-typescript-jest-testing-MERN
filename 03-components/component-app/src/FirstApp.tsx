//Using typescript types instead of proptypes
type NewMessage = {
    message: string,
    title: string
}

type Properties = {
    title?: string,
    subtitle: string
}

//Homework, return value from function
const returnMessage = (): NewMessage => {
    return {
        message: 'Hello world',
        title: 'A simple message'
    }
}

export const FirstApp = ({ title='Probe title', subtitle}: Properties) => {
    return (
        <>
            <h1 data-testid='test-title'>{title}</h1>
            <h2 data-testid='test-subtitle'>{subtitle}</h2>
            <h3>{JSON.stringify(returnMessage())}</h3>
        </>
    )
}

