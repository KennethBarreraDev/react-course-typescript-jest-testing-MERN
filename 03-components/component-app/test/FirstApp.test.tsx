import { render } from "@testing-library/react";
import { FirstApp } from "../src/FirstApp.tsx"; // Importa tu componente correctamente

describe("FirstApp Component", () => {
  test("Should render subtitle by test id", () => {
    const subtitle = 'First paragraph';
    const { getByText, getByTestId } = render(<FirstApp subtitle={subtitle} />);
    expect(getByText(subtitle)).toBeTruthy();

    expect(getByTestId('test-subtitle').innerHTML).toContain(subtitle)

    // const h2 = container.querySelector('h2')
    // expect(h2?.innerHTML).toContain(subtitle)
  });

  test("Should render subtitle by test id", () => {
    const title = 'This is a title';
    const subtitle = 'First paragraph';

    const { getByText, getByTestId } = render(<FirstApp subtitle={subtitle} title={title} />);
    expect(getByText(title)).toBeTruthy();
    expect(getByTestId('test-title').innerHTML).toContain(title)
  });
});
