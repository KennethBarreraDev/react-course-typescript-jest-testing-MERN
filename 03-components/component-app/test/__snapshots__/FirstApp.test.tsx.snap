// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FirstApp Component should match snapshot 1`] = `
<div>
  <h1>
    Probe title
  </h1>
  <h2>
    First paragraph
  </h2>
  <h3>
    {"message":"Hello world","title":"A simple message"}
  </h3>
</div>
`;
